router-tests/lifecycle/supervisor_test.go

```diff
@@ -37,10 +37,15 @@ func TestRouterSupervisor(t *testing.T) {
 	xEnv.RouterSupervisor.Stop()
 	xEnv.Shutdown()
 
-	// Let everything settle
-	time.Sleep(1 * time.Second)
+	for {
```

Comment:
My question here would be why is the server not properly shutdown after `xEnv.Shutdown()` Can we change the code to wait for graceful shutdown? e..g by introducing a close method that waits until the goroutine has been shutdown.

Reply:
It does normally, it's that we aren't actually blocking on the router being shut down here because of the supervisor being in the unchecked goroutine above. If this weren't a test and we could block on `RouterSupervisor.Start()` it would be fine.

`RouterSupervisor.Stop()` is non-blocking, it's just a wrapper for sending a message to the stop channel that will cause the other goroutine to shut down.

Could potentially have blocking stop? It would only be used in this test

Reply:
Why can't we implement the test in a way that it respect a graceful shutdown of the router? Can't you wait until Start unblocked?

Reply:
I can, done in latest commit